api/go.mod

```diff
@@ -20,13 +20,13 @@ require (
 	github.com/kr/text v0.2.0 // indirect
 	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
 	github.com/modern-go/reflect2 v1.0.2 // indirect
-	golang.org/x/net v0.18.0 // indirect
+	golang.org/x/net v0.19.0 // indirect
 	golang.org/x/text v0.14.0 // indirect
 	gopkg.in/check.v1 v1.0.0-20201130134442-10cb98267c6c // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
-	k8s.io/klog/v2 v2.100.1 // indirect
-	k8s.io/utils v0.0.0-20230726121419-3b25d923346b // indirect
+	k8s.io/klog/v2 v2.110.1 // indirect
```

Comment:
This is not Ok, 110 breaks backwards compatibly

Reply:
Hu, I couldn't find an obvious commit in https://github.com/kubernetes/klog/compare/v2.100.1...v2.110.1 that would do that.

Reply:
It appears to not break backwards compatibility when updated in conjunction with logr to v1.3.0. As my helm-controller build (which contains this bump) is running just fine.

Reply:
We could force controller-runtime to the new version if we bump logr also in okg/runtime, the version we are using is on klog 100 and logr 1.2 https://github.com/kubernetes-sigs/controller-runtime/blob/v0.16.3/go.mod

Reply:
I can look into this in combination with a weird issue I observed earlier when `log-level` is set to `trace`:

```
{"level":"Level(-2)","ts":"2023-12-11T10:45:58.333Z","msg":"getting last revision of \"podinfo\"","controller":"helmrelease","controllerGroup":"helm.toolkit.fluxcd.io","controllerKind":"HelmRelease","HelmRelease":{"name":"podinfo","namespace":"podinfo"},"namespace":"podinfo","name":"podinfo","reconcileID":"daa23d2f-7b8a-4428-a61e-a816fe7d2846"}
```

Reply:
Can someone please clarify on the backwards-incompatibility we're talking about here? The code compiles and tests fine so is it about the actual output it produces that's different with the new version?

Reply:
"Backwards compatibility being" that klog v2.110 is only compatible with logr >=v1.3.0

Reply:
The backwards-incompatibility is about klog - logr, controller-runtime comes with logr 1.2 which doesn't work with klog 110. As I said above, we could force logr in pkg/runtime.